lib/modules/manager/woodpecker/extract.ts

```diff
@@ -5,6 +5,18 @@ import { getDep } from '../dockerfile/extract';
 import type { ExtractConfig, PackageFileContent } from '../types';
 import type { WoodpeckerConfig } from './types';
 
+function woodpeckerVersionDecider(
+  woodpeckerConfig: WoodpeckerConfig
+): keyof WoodpeckerConfig {
+  if ('steps' in woodpeckerConfig) {
+    return 'steps';
+  } else if ('pipeline' in woodpeckerConfig) {
+    return 'pipeline';
+  }
+
+  throw new Error('No matching pipeline');
```

Comment:
better just return null and check it on caller. performance is better than exception handling